src/arrayMethodSort.js

```diff
@@ -5,8 +5,33 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let str = '';
+
+    const compare = compareFunction || defaultCompare;
+
+    while (str !== this.toString()) {
+      str = this.toString();
+
+      for (let i = 0; i < this.length - 1; i += 1) {
+        const current = this[i];
+        const next = this[i + 1];
+
+        if (compare(current, next) > 0) {
```

Comment:
here you can change 
`compare(current, next)`
to
`compareFunction(current, next)`